core/modules/acl.cc

```diff
@@ -0,0 +1,65 @@
+#include "acl.h"
+
+#include <rte_ether.h>
+#include <rte_ip.h>
+#include <rte_lpm.h>
+#include <rte_udp.h>
+
+#include <string>
+
+const Commands ACL::cmds = {
+    {"add", "ACLArg", MODULE_CMD_FUNC(&ACL::CommandAdd), 0},
+    {"clear", "EmptyArg", MODULE_CMD_FUNC(&ACL::CommandClear), 0}};
+
+pb_error_t ACL::Init(const bess::pb::ACLArg &arg) {
+  for (const auto &rule : arg.rules()) {
+    ACLRule new_rule = {.src_ip = CIDRNetwork(rule.src_ip()),
+                        .dst_ip = CIDRNetwork(rule.dst_ip()),
+                        .src_port = static_cast<uint16_t>(rule.src_port()),
+                        .dst_port = static_cast<uint16_t>(rule.dst_port()),
+                        .established = rule.established(),
+                        .drop = rule.drop()};
+    rules_.push_back(new_rule);
+  }
+  return pb_errno(0);
+}
+
+pb_cmd_response_t ACL::CommandAdd(const bess::pb::ACLArg &arg) {
+  pb_cmd_response_t response;
+  set_cmd_response_error(&response, Init(arg));
+  return response;
+}
+
+pb_cmd_response_t ACL::CommandClear(const bess::pb::EmptyArg &) {
+  rules_.clear();
+  return pb_cmd_response_t();
+}
+
+void ACL::ProcessBatch(bess::PacketBatch *batch) {
+  gate_idx_t out_gates[bess::PacketBatch::kMaxBurst];
+  int cnt = batch->cnt();
+  for (int i = 0; i < cnt; i++) {
+    bess::Packet *pkt = batch->pkts()[i];
+
+    struct ether_hdr *eth = pkt->head_data<struct ether_hdr *>();
+    struct ipv4_hdr *ip = reinterpret_cast<struct ipv4_hdr *>(eth + 1);
+    struct udp_hdr *udp = reinterpret_cast<struct udp_hdr *>(ip + 1);
```

Comment:
Do we need a check for options here?  If there are options, then ip+1 won't point at the UDP header.